fellow-submissions/peter-rose.md

```diff
@@ -0,0 +1,148 @@
+# Problem Set 2.4 - Scope
+
+**1. How do variables declared with `let`, `const`, and `var` differ? Be sure to touch on _scope_, _reassignment_, and _hoisting_ for each.**
+
+<!---->
+   Firstly, const cannot change its value. Thats the purpose of it so that variable will never be able to reassign to a new value. Unlike
+   let and var which are able to be reassigned.
+   
+   Secondly, let and const hoist but you can't access them before the declaration is read by the console. Check out Questions 6 & 7
+   in this .md file for a good example of this.
+   
+   Thirdly, var can be accessible anywhere in function but let and const can only be accessible inside the block where you first declare them.
+
+**2. What will the following code log? Why?**
+    ```javascript
+    let a = 'outer';
+
+    function testScope() {
+      let a = 'inner';
+      console.log(a);
+    }
+
+    console.log(a);
+    testScope();
+    console.log(a);
+    ```
+    <!---->
+    console.log(a) will ALWAYS log as outer. The function called on line 15 is refering to the variable 'a' thats initialized within itself
+    so it'll always refer to that value which is inner.
+    //ouput    outer --> inner --> outer
+
+**3. What will the following code log? Why?**
+    ```javascript
+    function hello() {
+      let a = 'hello';
+    }
+
+    hello();
+    console.log(a);
+    ```
+    <!---->
+    Code will result in an error, because a is not defined. this is because let has a block scope so the value of a cannot be 
+    accessed because the "let" keyword is only accessible inside the block where they are declared, 
+    just like "const" and unlike "var".
+
+**4. What does the following code log? Why?**
+    ```javascript
+    console.log(a);
+
+    var a = 1;
+    ```
+    <!---->
+    The code logs as undefined. This is because var may have a normal scope so due to hoisting the console knows a 
+    variable named a does exist but it is called upon because that variable is initialized, and cannot retrieve the value.
+  
+**5. What does the following code log? Why?**
+    ```javascript
+    console.log(a);
+
+    let a = 1;
+    ```
+    <!---->
+    The code results in an error saying that 'a' is not defined. This is signaificantly different from the previous example because
```

Comment:
> The code results in an error saying that 'a' is not defined.
This is incorrect. Run this in the console and see what type of error and message you get. It has less to do with block scope but more to do with how `let` variables are hoisted. Try this one again.